### PR TITLE
Internship update

### PR DESCRIPTION
**Pull request by uttamthummala**
- Updated To accomodate Internships
- Added Tests
- Fixed Mailer
- Added Addissue Endpoint
- Fixed Logic Issues StudentAcceptOffer Endpoint

### CODE REVIEW

File: CDC_Backend/APIs/admin.py
Comment (uttamthummala) [inline CDC_Backend/APIs/admin.py:1]: Registered InternshipApplication and Internship Models

File: CDC_Backend/APIs/studentViews.py [hunk 1]
```diff
@@ -277,19 +322,86 @@ def getContributorStats(request, id, email, user_type):
...
     try:
-        company_id = request.data['id']
-        student_id=request.data['profileInfo']['id']
-        offer_status = request.data['offerStatus']
-        placement_application=PlacementApplication.objects.get(placement=company_id,student=student_id)
-        placement_application.offer_accepted=offer_status
-        placement_application.save()
-        return Response({'action': "Accept Offer", 'message': "Updated Offer Status"},
+        company_id = request.data[OPENING_ID]
+        #student_id=request.data['profileInfo']['id']             //check this once
```
Comment (uttamthummala) [inline CDC_Backend/APIs/studentViews.py:330]: As The endpoint is Specific to a Student id can be extracted directly without getting it from request

File: CDC_Backend/APIs/adminViews.py [hunk 2]
```diff
@@ -110,13 +145,25 @@ def updateOfferAccepted(request, id, email, user_type):
     try:
         data = request.data
         offer_accepted = data[OFFER_ACCEPTED]
-        opening = get_object_or_404(Placement, pk=data[OPENING_ID])
+        if  OPENING_TYPE  in data:
+            opening_type= data[OPENING_TYPE]
+        else:
+            opening_type= "Placement"
+        if DEADLINE_DATETIME in data:
```
Comment (uttamthummala) [inline CDC_Backend/APIs/adminViews.py:152]: Automatically Creating a deadline when accepting offer if the field is not provided in request .if provided using the given field

File: CDC_Backend/APIs/utils.py [hunk 3]
```diff
@@ -194,7 +198,7 @@ def sendEmail(email_to, subject, data, template, attachment_jnf_response=None):
         else:
             recipient_list = [str(email_to), ]
 
-        msg = EmailMultiAlternatives(subject, text_content, email_from, recipient_list)
+        msg = EmailMultiAlternatives(subject, text_content, email_from,None,bcc=recipient_list)
```
Comment (karthikmurakonda) [inline CDC_Backend/APIs/utils.py:201]: Test this

**Review by karthikmurakonda — APPROVED** [review]
Thank you very much uttam!

**Pull request merged**